server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -773,24 +649,8 @@ private TSExecuteStatementResp internalExecuteQueryStatement(String statement,
 
       return resp;
     } catch (Exception e) {
-      if (e instanceof NullPointerException) {
-        logger.error("{}: Internal server error: ", IoTDBConstant.GLOBAL_DB_NAME, e);
-      } else {
-        logger.warn("{}: Internal server error: ", IoTDBConstant.GLOBAL_DB_NAME, e);
-      }
-      if (queryId != -1) {
-        try {
-          releaseQueryResource(queryId);
-        } catch (StorageEngineException ex) {
-          logger.warn("Error happened while releasing query resource: ", ex);
-        }
-      }
-      Throwable cause = e;
-      while (cause.getCause() != null) {
-        cause = cause.getCause();
-      }
-      return RpcUtils
-          .getTSExecuteStatementResp(TSStatusCode.INTERNAL_SERVER_ERROR, cause.getMessage());
+      releaseQueryResourceNoExceptions(queryId);
```

Comment:
It's better to put it in the `finally` block...

Reply:
why?

Reply:
Oh.. Ignore it.. :D